assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -68,35 +68,38 @@ const baseActions = {
 	 */
 	*syncPublicationOnboardingState() {
 		const registry = yield commonActions.getRegistry();
-		const connected = yield commonActions.await(
-			registry
-				.resolveSelect( CORE_MODULES )
-				.isModuleConnected( MODULE_SLUG )
-		);
 
-		// If the module is not connected, do not attempt to sync the onboarding state.
-		if ( ! connected ) {
+		const hasPublicationIDChanged = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.hasSettingChanged( 'publicationID' );
```

Comment:
This will return `false` if settings haven't been loaded yet, but also, this selector won't invoke the resolver for `getSettings` so we need to resolve select those first to ensure the state is loaded before calling this.

Reply:
Thank you for addressing this, @aaemnnosttv !